src/pages/Homepage/Homepage.vue

```diff
@@ -29,69 +26,30 @@ export default {
 	},
 	data() {
 		return {
-			mgPromoExp: { id: null, version: null },
-			doubleArrowButtonExp: { id: null, version: null },
-			promoEnabled: false,
-			holidayModeEnabled: false,
+			promoEnabled: false
 		};
 	},
 	inject: ['apollo'],
 	apollo: {
 		query: promoQuery,
+		variables() {
+			return {
+				contentType: 'uiSetting',
+				contentKey: 'ui-homepage-promo',
+			};
+		},
 		preFetch: true,
 		result({ data }) {
-			this.holidayModeEnabled = settingEnabled(
-				data,
-				'general.holiday_enabled.value',
-				'general.holiday_start_time.value',
-				'general.holiday_end_time.value'
-			);
-
+			// returns the contentful content of the uiSetting key ui-homepage-promo or empty object
+			// it should always be the first and only item in the array, since we pass the variable to the query above
+			const uiPromoSetting = _get(data, 'contentfulCMS.items', []).find(item => item.key === 'ui-homepage-promo'); // eslint-disable-line max-len
 			this.promoEnabled = settingEnabled(
-				data,
-				'general.promo_enabled.value',
-				'general.promo_start_time.value',
-				'general.promo_end_time.value'
-			);
-
-			this.lendingRewardOffered = _get(data, 'shop.lendingRewardOffered');
-		}
-	},
-	created() {
-		// --------------------
-		// Uncomment when the Billion To Women Campaign ends on Oct. 13th 2019
-		// ---------------------
-
-		// get exp assignment for monthly good promo
-		// this.mgPromoExp = this.apollo.readFragment({
-		// 	id: 'Experiment:mg_promo',
-		// 	fragment: experimentVersionFragment,
-		// }) || {};
-
-		// get exp assignment for double arrow button experiment
-		this.doubleArrowButtonExp = this.apollo.readFragment({
-			id: 'Experiment:double_arrow_button',
-			fragment: experimentVersionFragment,
-		}) || {};
-	},
-	mounted() {
-		// --------------------
-		// Uncomment when the Billion To Women Campaign ends on Oct. 13th 2019
-		// ---------------------
-
-		// if (this.mgPromoExp.version !== null) {
-		// 	this.$kvTrackEvent(
-		// 		'Lending',
-		// 		'EXP-CASH-129-Sept2019',
-		// 		this.mgPromoExp.version === 'shown' ? 'b' : 'a'
-		// 	);
-		// }
-		if (this.doubleArrowButtonExp.version !== null) {
-			this.$kvTrackEvent(
-				'Homepage',
-				'EXP-CASH-1313-Oct2019',
-				this.doubleArrowButtonExp.version === 'shown' ? 'b' : 'a'
+				uiPromoSetting,
+				'active',
+				'startDate',
+				'endDate'
 			);
+			// this.promoEnabled = uiPromoSetting.active; // use this to bypass date during dev
```

Comment:
To test the Contentful integration prior to Giving Tuesday, uncomment this line and toggle the ui-homepage-promo active setting within Contentful development space.

Reply:
It seems to me that this is prone to error, forgetting to uncomment this line, etc. We should uncomment this line. If you want to enable or disable it, you can change the setting in the contentful development space, that is the reason why development exists. It will also allow us to test this on dev.kiva.org

Reply:
Good points, thanks.

Reply:
I've removed this line, to test you can change the date range and active setting in Contentful development space.